api/tests/acceptance/application/session/session-controller-get-jury-session_test.js

```diff
@@ -0,0 +1,90 @@
+const { expect, databaseBuilder, generateValidRequestAuthorizationHeader, insertUserWithRolePixMaster } = require('../../../test-helper');
+const createServer = require('../../../../server');
+
+describe('Acceptance | Controller | session-controller-get-jury-session', () => {
+
+  let server, options;
+
+  beforeEach(async () => {
+    server = await createServer();
+    await insertUserWithRolePixMaster();
+  });
+
+  describe('GET /api/jury/sessions/{id}', () => {
+    let expectedJurySession;
+
+    beforeEach(() => {
+      const assignedCertificationOfficerId = databaseBuilder.factory.buildUser({ firstName: 'Pix', lastName: 'Doe' }).id;
+      const certificationCenter = databaseBuilder.factory.buildCertificationCenter({ type: 'SCO' });
+      expectedJurySession = databaseBuilder.factory.buildSession({
+        assignedCertificationOfficerId,
+        certificationCenterId: certificationCenter.id,
+        certificationCenter: certificationCenter.name,
+      });
+      databaseBuilder.factory.buildSession();
+      options = {
+        method: 'GET',
+        url: `/api/jury/sessions/${expectedJurySession.id}`,
+      };
+
+      return databaseBuilder.commit();
+    });
+
+    context('when user is Pix Master', () => {
+      beforeEach(() => {
+        options.headers = { authorization: generateValidRequestAuthorizationHeader() };
+      });
+
+      it('should return a 200 status code response with JSON API serialized', async () => {
+        // when
+        const response = await server.inject(options);
+
+        // then
+        expect(response.statusCode).to.equal(200);
+        expect(response.result.data.type).to.equal('sessions');
+        expect(parseInt(response.result.data.id)).to.equal(expectedJurySession.id);
+        expect(response.result.data.attributes['certification-center-name']).to.equal(expectedJurySession.certificationCenter);
+        expect(response.result.data.attributes['certification-center-type']).to.equal('SCO');
+        expect(response.result.data.attributes['address']).to.equal(expectedJurySession.address);
+        expect(response.result.data.attributes['room']).to.equal(expectedJurySession.room);
+        expect(response.result.data.attributes['examiner']).to.equal(expectedJurySession.examiner);
+        expect(response.result.data.attributes['date']).to.equal(expectedJurySession.date);
+        expect(response.result.data.attributes['time']).to.equal(expectedJurySession.time);
```

Comment:
en vrai ça fait quand même beaucoup d'expect , si on a 2 ou 3 bons ça suffit pas à dire que les autres sont bon aussi ?

Reply:
C'est pas très grave qu'il y ait beaucoup d'expect. Là on s'assure que les données insérées dans le beforeEach, sont bien restituées, de bout en bout, dans les champs attendus.
Qu'entends t'on pas "trop" d'expect ? D'un point de vue perf et vitesse de déroulement du test, je pense que l'impact est inexistant. D'un point de vue de la lisibilité, why  not. Mais compromettre la couverture testée pour de la lisibilité... J'attends d'autres avis.